### DISCUSSION TITLE
next/image - how to convert to webp format?

### DISCUSSION THREAD

**Discussion by bushblade** [vercel/next.js#24592] [Help]
Hi all, I'm exploring NextJS and coming from Gatsby.

Is there a way using next/image to automatically convert jpg and png format images to webp like you can in Gatsby image?
I can't find anything related to this in the docs, other than next/image supports webp formats.
Is this possible?

**Comment by bobarros** [comment]
If the browser supports, it's going to be the first option:

_The Automatic Image Optimization allows for resizing, optimizing, and serving images in modern formats like WebP when the browser supports it_

Third paragraph here --> https://nextjs.org/docs/basic-features/image-optimization

**Reply by bushblade** [reply]
Hi, thanks for the reply.
So it's supposed to just convert if the browser supports it out of the box so to speak?
I'm testing in FireFox and checking the currentSrc property indicates it's still a jpg
![Screenshot_2021-04-29_13-42-12](https://user-images.githubusercontent.com/21976188/116552239-b9567300-a8f0-11eb-9378-da133aee03ce.png)
Same in Chromium too
![Screenshot_2021-04-29_13-43-27](https://user-images.githubusercontent.com/21976188/116552428-e99e1180-a8f0-11eb-8b2a-8a757750f838.png)

What am I missing?

**Reply by bobarros** [reply]
I see.

Go to the result in Chrome, save the image in some folder and you see the real format served.

![box](https://user-images.githubusercontent.com/34839745/116553075-1c83dd80-a8d0-11eb-8b62-d24ebdee6a0a.png)

**Reply by bushblade** [reply]
Ah yes you are correct.
I was expecting it to show in the currentSrc as a webp, like Gatsby image.
That's strange that it shows as a jpg isn't it? What's going on there?

**Reply by bobarros** [reply]
Because it's just the name of the original file that you're seen it. And then, you're assuming that this is the format.

![name](https://user-images.githubusercontent.com/34839745/116555599-f6ac0800-a8d2-11eb-8f6f-af428e86459f.png)

**Reply by bushblade** [reply]
currentSrc should give the url of the image currently being shown though? [MDN link](https://developer.mozilla.org/en-US/docs/Web/API/HTMLImageElement/currentSrc)
Like if I you just look at the img src in a Gatsby image, it will still show the jpg or png but the currentSrc shows that it is a webp that is currently showing.
So src and currentSrc are not necessarily the same thing.

**Reply by bobarros** [reply]
My man, I enjoy helping because it helps me to learn a lot. 

This is how deep I'm willing to go in this matter, though. 

Best of luck.

**Reply by bushblade** [reply]
No problem at all, thanks you've been very helpful.
I expect that currentSrc only applies to a srcSet and Next doesn't appear to be using that.
I may dig deeper but at least I see that it is working.

**Reply by vegerot** [reply]
How does the WebP conversion work?  Does it cache the images?  Where can I learn more about it than the single sentence

> Size Optimization: Automatically serve correctly sized images for each device, using modern image formats like WebP and AVIF.
